src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentPreprocessorDirectiveTest.cs

```diff
@@ -221,24 +221,18 @@ public void StartOfLine_04()
         AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
         AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
         CompileToAssembly(generated,
-            // x:\dir\subdir\Test\TestComponent.cshtml(2,9): error CS1040: Preprocessor directives must appear as the first non-whitespace character on a line
+            // x:\dir\subdir\Test\TestComponent.razor(2,9): error CS1040: Preprocessor directives must appear as the first non-whitespace character on a line
             // var x = #if true;
             Diagnostic(ErrorCode.ERR_BadDirectivePlacement, "#").WithLocation(2, 9),
-            // x:\dir\subdir\Test\TestComponent.cshtml(2,17): error CS1025: Single-line comment or end-of-line expected
-            // var x = #if true;
-            Diagnostic(ErrorCode.ERR_EndOfPPLineExpected, ";").WithLocation(2, 17),
-            // x:\dir\subdir\Test\TestComponent.cshtml(5,1): error CS0841: Cannot use local variable 'x' before it is declared
+            // x:\dir\subdir\Test\TestComponent.razor(5,1): error CS0841: Cannot use local variable 'x' before it is declared
             // x #endif;
             Diagnostic(ErrorCode.ERR_VariableUsedBeforeDeclaration, "x").WithArguments("x").WithLocation(5, 1),
-            // x:\dir\subdir\Test\TestComponent.cshtml(5,2): error CS1002: ; expected
+            // x:\dir\subdir\Test\TestComponent.razor(5,2): error CS1002: ; expected
             // x #endif;
             Diagnostic(ErrorCode.ERR_SemicolonExpected, "").WithLocation(5, 2),
-            // x:\dir\subdir\Test\TestComponent.cshtml(5,3): error CS1040: Preprocessor directives must appear as the first non-whitespace character on a line
-            // x #endif;
-            Diagnostic(ErrorCode.ERR_BadDirectivePlacement, "#").WithLocation(5, 3),
-            // x:\dir\subdir\Test\TestComponent.cshtml(5,9): error CS1025: Single-line comment or end-of-line expected
+            // x:\dir\subdir\Test\TestComponent.razor(5,3): error CS1040: Preprocessor directives must appear as the first non-whitespace character on a line
             // x #endif;
-            Diagnostic(ErrorCode.ERR_EndOfPPLineExpected, ";").WithLocation(5, 9));
+            Diagnostic(ErrorCode.ERR_BadDirectivePlacement, "#").WithLocation(5, 3));
```

Comment:
So we already have this error. Why do we need the Razor one?

If there are cases where the C# error is not reported, we probably should add C# codegen tests for them.

Reply:
> If there are cases where the C# error is not reported, we probably should add C# codegen tests for them.

Added `StartOfLine_12`.